src/agent/mod.rs

```diff
@@ -33,6 +34,10 @@ pub struct LoginOptions {
     ///
     /// If this field is empty, the current URL is used as a redirect URL.
     pub redirect_url: Option<Url>,
+    /// Keep the current url in the session
+    ///
+    ///Set this flag to true to store the current url for later retrieval in the session
+    pub keep_current_url_in_session: Option<bool>,
```

Comment:
I am not sure this needs to be `Option`, it looks like a default (false) would just work the same way.